lib/Dialect/AIEX/Transforms/AIEDMATasksToNPU.cpp

```diff
@@ -260,6 +273,40 @@ struct AIEDMATasksToNPUPass : AIEDMATasksToNPUBase<AIEDMATasksToNPUPass> {
         input_sizes[i] = (*dims)[j].getSize();
         input_strides[i] = (*dims)[j].getStride();
       }
+      if (dims->size() > 2) {
+        d2size = (target_model.isMemTile(tile.getCol(), tile.getRow()))
+                     ? (*dims)[2].getSize()
+                     : 0;
+      }
+      // if(failed(verifyPaddingDims(target_model, tile.getCol(), tile.getRow(),
+      // dims, padDims))) return failure();
+      if (padDims.has_value()) {
+        if (!target_model.isMemTile(tile.getCol(), tile.getRow()))
+          return bd_op->emitOpError()
+                 << "Padding is only supported by memtile dma bds.";
+        if (padDims->size() > dims->size())
+          return bd_op->emitOpError()
+                 << "Mismatch number of dimensions between padding(s)"
+                 << " and wrap(s) and stride(s).";
+      }
+
+      if (target_model.isMemTile(tile.getCol(), tile.getRow()) &&
+          channelDir == AIE::DMAChannelDir::MM2S) {
+        if (padDims) {
+          for (size_t i = 0; i < padDims->size(); i++) {
+            int j = padDims->size() - i - 1;
+            padBefore[i] = (*padDims)[j].getConstPadBefore();
+            padAfter[i] = (*padDims)[j].getConstPadAfter();
+          }
+          for (size_t i = padDims->size(); i < dims->size(); i++) {
+            padBefore[i] = 0;
+            padAfter[i] = 0;
+          }
+        }
+      } else if (padDims) {
```

Comment:
This else-if can then just be an else to the if which checks if we are on the MM2S channel.